proc_scripts/side_by_side.py

```diff
@@ -0,0 +1,48 @@
+from pyspecdata import *
+from sympy import symbols
+def expand_limits(thisrange,s):
+    thisrange = list(thisrange)
+    full_range = s.getaxis('t2')[r_[0,-1]]
+    retval = array([thisrange[j] if thisrange[j] is not
+            None else full_range[j] for j in [0,-1]])
+    print(repr(retval))
+    m = mean(retval)
+    s = m-retval
+    retval = 3*s+m
+    sgn = [-1,1] # greater than or less than
+    return tuple(full_range[j] if
+            retval[j]*sgn[j] > full_range[j]*sgn[j]
+            else
+            retval[j] for j in range(2))
+def draw_limits(thisrange,s):
+    full_range = s.getaxis('t2')[r_[0,-1]]
+    dw = diff(s.getaxis('t2')[:2]).item()
+    print("I find the full range to be",full_range)
+    sgn = [-1,1] # add or subtract
+    pairs = [[thisrange[j],full_range[j]+0.5*dw*sgn[j]] for j in range(2)]
+    pairs[0] = pairs[0][::-1] # flip first
+    my_xlim = gca().get_xlim() # following messes w/ xlim for some reason
+    for j in pairs:
+        if None not in j:
+            print("drawing a vspan at",j)
+            axvspan(j[0],j[1],color='w',alpha=0.5,linewidth=0)
+    gca().set_xlim(my_xlim)
+class fl_mod (figlist_var):
+    def side_by_side(self,plotname,s,thisrange):
```

Comment:
I'm not convinced this is the best way to do this -- here, you're creating a new class that inherits from `figlist_var` and adding the side_by_side function to it.  It might be better to define the `side_by_side` function here, and then make the inherited class inside whatever script you are using -- I think this should work:
```
class fl_mod(figlist_var):
    side_by_side = side_by_side
```

Reply:
I'm not sure what you mean let's discuss at next meeting

Reply:
please look into inheritance in classes.  Here is a [link to the relevant section in the Langtangen book](http://hplgit.github.io/primer.html/doc/pub/oo/._oo-bootstrap001.html#sec:oo:inheritance)

Reply:
done